src/mixins/filing-template-mixin.ts

```diff
@@ -321,9 +322,18 @@ export default class FilingTemplateMixin extends DateMixin {
       filing.restoration.expiry = this.getRestoration.expiry
     }
 
-    // Apply NR / business name / business type change to filing
-    if (this.getNameRequestNumber || this.hasBusinessNameChanged || this.hasBusinessTypeChanged) {
-      filing.restoration.nameRequest = this.getNameRequest
+    // Set Name Request object for a full restoration only
+    if (this.isLimitedRestorationToFull) {
+      // Apply NR / business name change to filing
+      if (this.getNameRequestNumber || this.hasBusinessNameChanged) {
+        filing.restoration.nameRequest = this.getNameRequest
+      } else {
+        // Otherwise save default data
+        filing.restoration.nameRequest = {
+          legalName: this.getEntitySnapshot.businessInfo.legalName,
+          legalType: this.getEntitySnapshot.businessInfo.legalType
+        }
+      }
     }
```

Comment:
Theoretically, `getNameRequestNumber` and `hasBusinessNameChanged` were never True for a restoration extension (since the UI component is not rendered). Also, we don't change business type in a restoration extension/conversion.

So the new code is more strict but should be equivalent.